client/src/main/scala/com.crobox.clickhouse/internal/InternalExecutorActor.scala

```diff
@@ -17,11 +18,17 @@ class InternalExecutorActor(override protected val config: Config)
   override implicit val system: ActorSystem = context.system
 
   override def receive = {
-    case Execute(uri: Uri, query: String) =>
+    case Execute(uri: Uri, Some(query)) =>
       val eventualResponse = executeRequest(Future.successful(uri), query)
-      eventualResponse.map(response => response.split("\n").toSeq) pipeTo sender
+      splitResponse(eventualResponse) pipeTo sender
+    case Execute(uri: Uri, None) =>
+      val request = HttpRequest(method = HttpMethods.GET, uri = uri)
+      splitResponse(handleResponse(singleRequest(request), "health check", uri)) pipeTo sender
```

Comment:
Is the `singleRequest` using a different queue? So issue #42 is also fixed then?

Reply:
This uses an internal queue for health/cluster member scanning.